### DISCUSSION TITLE
question about upbge

### DISCUSSION THREAD

**Discussion by drcucaracha** [UPBGE/upbge#1588] [Ideas/Features for UPBGE 0.2.5+]
hello I would like to solve a question I have, is it possible to convert the upbge (eevee) renderer as armory3d does but instead of using blender to make it work, use godot engine? because the user interface is much clearer but the 3d renderer is not very good and it is poor in performance

**Comment by lordloki** [comment]
UPBGE 0.2.5 doesn't use eevee as renderer, it uses an own renderer. UPBGE 0.3 uses eevee renderer and it is not possible to replace for other one due to its integration in Blender. Eevee will improve in the next versions (rewrite to deferred and Vulkan).

**Reply by HolandonHdunto** [reply]
upbge 0.3 despite having amazing graphics is not suitable for making weak pc games, especially big and complex games like this guy https://youtu.be/1vdQfNR7AkM or https://youtu.be/RxHah4uXOwU, it uses upbge 0.2. 5. I think cube mapping is as good as eevee, the problem is that upbge 0.2.5 glsl nodes are limited and don't have variables or how to capture object position and rotation, actually if only I could introduce variables or properties on glsl nodes would suffice. Generally those who use upbge 0.2.5 usually use the lamps as variables, the problem is that it does not work in groups, and has a limited value.

**Comment by BluePrintRandom** [comment]
https://github.com/ufz/tetgen
https://www.gdcvault.com/play/1015312/Light-Probe-Interpolation-Using-Tetrahedral

check these two out @Holandon

we define a walkable area - place probes near light sources - bake the
probes and the form tehtredron from them, and interpolate them per agent/
actor
